tool_registry/manifest.py

```
@@ -227,19 +227,27 @@ def resolve_config(base_dir: Path) -> dict[str, Any]:
                     if js_entry and node_path:
-                        cmd[0] = js_entry
-                        cmd.insert(0, node_path)
+                        # Rebuild the command to avoid remnants from half-resolved
+                        # or polluted configs (e.g. ['node', 'old-bin', ...]).
+                        # We keep the flags (everything after the first element).
+                        flags = original_cmd[1:]
+                        if original_cmd[0] == "node" and len(original_cmd) > 1:
+                            # If it was already prefixed with node, flags start at index 2
+                            flags = original_cmd[2:]
+
+                        section["command"] = [node_path, js_entry] + flags
```

Ruff (0.15.12)

[warning] Consider `[node_path, js_entry, *flags]` instead of concatenation

Replace with `[node_path, js_entry, *flags]`

(RUF005)